### PR TITLE
Feat : seoUpgrade for kuzzle documentation

### PR DESCRIPTION
**Pull request by thomas-mauran**
##  Metadata decription title and keywords update to get better google SEO. 

Adding this to most of the markdown doc headers with this type of header example :
```
title:  General Purpose Backend | Kuzzle Introduction | Guide | Core
meta:
  name: description
    content: Kuzzle main concepts and features overview
  name: keywords
    content: Kuzzle, Documentation, kuzzle write pluggins, General purpose backend
```

### How should this be manually tested?

  - Step 1 : after the merge check if the metadatas and title are updated on the documentation pages using a web browser extention
  - Step 2 : if updates aren't applied try to reindex the website using google search console
  - Step 3 : if it's good but v1 is still on top need to find new solutions (maybe around s3, google console or canonicals)

### Other changes

added the following on type: 'branch' to avoid those type of blank links : https://docs.kuzzle.io/paas-console/1/api/
```
<Redirect to="folder" />
```

### CODE REVIEW

**Comment by kuzzle** [comment]
<table>
  <thead>
    <tr>
      <th width="50"></th>
      <th width="100%" data-danger-table="true">Fails</th>
    </tr>
  </thead>
  <tbody><tr>
      <td>:no_entry_sign:</td>
      <td>Found some issues !</td>
    </tr>
  </tbody>
</table>



## Executed Checks

### :x: Branch Merge Check

Merging "feat/seoUpgrade" into "master" is not allowed.
If you want to allow this PR to be merged into "master", add the following line to your PR body:


```
ci-allow-merge-into: master
```

### :x: Changelog Tag Check

Pull Requests must have a `changelog:XXX` label.
If you want to allow this PR to not have a changelog tag, add the following line to the PR body:


```
ci-no-changelog-tag
```

### :heavy_check_mark: PR Body Check


<p align="right">
  Generated by :no_entry_sign: <a href="https://danger.systems/js">dangerJS</a> against eab08ae3e15018a94b1185c50eda1071fbf401c4
</p>

File: doc/2/guides/advanced/index.md [hunk 1]
```diff
@@ -1,9 +1,13 @@
 ---
 code: false
 type: branch
-title: Advanced
-description: Kuzzle API advanced features and concepts
 order: 500
+title:  Kuzzle Advanced | Guide | Core
```
Comment (alexandrebouthinon) [inline doc/2/guides/advanced/index.md:5]:
```suggestion
title: Kuzzle Advanced | Guide | Core
```

File: doc/2/guides/develop-on-kuzzle/event-system/index.md [hunk 2]
```diff
@@ -1,11 +1,14 @@
 ---
 code: false
 type: page
-title: Event System
-description: Interact with Internal Events
 order: 300
+title:  Event System | Develop on Kuzzle | Guide | Core
```
Comment (alexandrebouthinon) [inline doc/2/guides/develop-on-kuzzle/event-system/index.md:5]:
```suggestion
title: Event System | Develop on Kuzzle | Guide | Core
```

File: doc/2/guides/develop-on-kuzzle/index.md [hunk 3]
```diff
@@ -1,11 +1,14 @@
 ---
 code: false
 type: branch
-title: Develop on Kuzzle
-description: Write custom backend code for Kuzzle
 order: 400
+title:  Develop on Kuzzle | Guide | Core
```
Comment (alexandrebouthinon) [inline doc/2/guides/develop-on-kuzzle/index.md:5]:
```suggestion
title: Develop on Kuzzle | Guide | Core
```

File: doc/2/guides/getting-started/customize-api-behavior/index.md [hunk 4]
```diff
@@ -1,11 +1,16 @@
 ---
 code: false
 type: page
-title: Customize the API Behavior
-description: Use the fine-grained middleware-like system
 order: 800
+title:  Customize the API Behavior | Kuzzle Getting Started | Guide | Core
```
Comment (alexandrebouthinon) [inline doc/2/guides/getting-started/customize-api-behavior/index.md:5]:
```suggestion
title: Customize the API Behavior | Kuzzle Getting Started | Guide | Core
```

File: doc/2/guides/getting-started/deploy-your-application/index.md [hunk 5]
```diff
@@ -1,9 +1,13 @@
 ---
 code: false
 type: page
-title: Deploy your Application
-description: Deploy your Kuzzle application on a remote server
 order: 900
+title:  Deploy your Application | Kuzzle Getting Started | Guide | Core
```
Comment (alexandrebouthinon) [inline doc/2/guides/getting-started/deploy-your-application/index.md:5]:
```suggestion
title: Deploy your Application | Kuzzle Getting Started | Guide | Core
```

File: doc/2/guides/getting-started/index.md [hunk 6]
```diff
@@ -1,10 +1,13 @@
 ---
 code: false
 type: branch
-title: Getting Started
-description: Learn how to use Kuzzle to develop your backend application
 order: 200
+title:  Kuzzle Getting Started | Guide | Core
```
Comment (alexandrebouthinon) [inline doc/2/guides/getting-started/index.md:5]:
```suggestion
title: Kuzzle Getting Started | Guide | Core
```

File: doc/2/guides/getting-started/run-kuzzle/index.md [hunk 7]
```diff
@@ -1,9 +1,13 @@
 ---
 code: false
 type: page
-title: Run Kuzzle
-description: Run your first Kuzzle application
 order: 100
+title:  Run Kuzzle | Kuzzle Getting Started | Guide | Core
```
Comment (alexandrebouthinon) [inline doc/2/guides/getting-started/run-kuzzle/index.md:5]:
```suggestion
title: Run Kuzzle | Kuzzle Getting Started | Guide | Core
```

File: doc/2/guides/getting-started/set-up-permissions/index.md [hunk 8]
```diff
@@ -1,11 +1,16 @@
 ---
 code: false
 type: page
-title: Set up Permissions
-description: Define user rights
 order: 300
+title:  Set up Permissions | Kuzzle Getting Started | Guide | Core
```
Comment (alexandrebouthinon) [inline doc/2/guides/getting-started/set-up-permissions/index.md:5]:
```suggestion
title: Set up Permissions | Kuzzle Getting Started | Guide | Core
```

File: doc/2/guides/getting-started/subscribe-realtime-notifications/index.md [hunk 9]
```diff
@@ -1,11 +1,14 @@
 ---
 code: false
 type: page
-title: Subscribe to Realtime Notifications
-description: Use the Realtime Engine to subscribe to database change
 order: 500
+title:  Subscribe to Realtime Notifications | Kuzzle Getting Started | Guide | Core
```
Comment (alexandrebouthinon) [inline doc/2/guides/getting-started/subscribe-realtime-notifications/index.md:5]:
```suggestion
title: Subscribe to Realtime Notifications | Kuzzle Getting Started | Guide | Core
```

File: doc/2/guides/getting-started/write-application/index.md [hunk 10]
```diff
@@ -1,9 +1,13 @@
 ---
 code: false
 type: page
-title: Write an Application
-description: Discover the framework capabilities
 order: 600
+title:  Write an Application | Kuzzle Getting Started | Guide | Core
```
Comment (alexandrebouthinon) [inline doc/2/guides/getting-started/write-application/index.md:5]:
```suggestion
title: Write an Application | Kuzzle Getting Started | Guide | Core
```

File: doc/2/guides/index.md [hunk 11]
```diff
@@ -2,8 +2,11 @@
 code: false
 type: root
 order: 0
-title: Guides
-description: Kuzzle v2.x Guides
+title:  What is kuzzle | Guides | Core
```
Comment (alexandrebouthinon) [inline doc/2/guides/index.md:5]:
```suggestion
title: What is kuzzle | Guides | Core
```

File: doc/2/guides/introduction/general-purpose-backend/index.md [hunk 12]
```diff
@@ -1,9 +1,13 @@
 ---
 code: false
 type: page
-title: General Purpose Backend
-description: Kuzzle main concepts and features overview
 order: 200
+title:  General Purpose Backend | Kuzzle Introduction | Guide | Core
```
Comment (alexandrebouthinon) [inline doc/2/guides/introduction/general-purpose-backend/index.md:5]:
```suggestion
title: General Purpose Backend | Kuzzle Introduction | Guide | Core
```

File: doc/2/guides/main-concepts/api/index.md [hunk 13]
```diff
@@ -1,9 +1,13 @@
 ---
 code: false
 type: page
-title: API
-description: Discover Kuzzle API usage and formats
 order: 100
+title:  Kuzzle API | Main Concepts | Guide | Core
```
Comment (alexandrebouthinon) [inline doc/2/guides/main-concepts/api/index.md:5]:
```suggestion
title: Kuzzle API | Main Concepts | Guide | Core
```

File: doc/2/guides/main-concepts/authentication/index.md [hunk 14]
```diff
@@ -1,9 +1,13 @@
 ---
 code: false
 type: page
-title: Authentication
-description: Authenticate your users through the multi-strategy system
 order: 500
+title:  Authentication | Main Concepts | Guide | Core
```
Comment (alexandrebouthinon) [inline doc/2/guides/main-concepts/authentication/index.md:5]:
```suggestion
title: Authentication | Main Concepts | Guide | Core
```

File: doc/2/guides/main-concepts/data-storage/index.md [hunk 15]
```diff
@@ -1,9 +1,13 @@
 ---
 code: false
 type: page
-title: Data Storage
-description: Understand how works the underlying document storage engine
 order: 200
+title:  Data Storage | Main Concepts | Guide | Core
```
Comment (alexandrebouthinon) [inline doc/2/guides/main-concepts/data-storage/index.md:5]:
```suggestion
title: Data Storage | Main Concepts | Guide | Core
```

File: doc/2/guides/write-protocols/context/index.md [hunk 16]
```diff
@@ -2,6 +2,12 @@
 code: false
 type: branch
 order: 200
-title: Protocol Framework
-description: Protocol Framework Reference
+title:  Protocol Framework | Write protocols | Guides | Core
```
Comment (alexandrebouthinon) [inline doc/2/guides/write-protocols/context/index.md:5]:
```suggestion
title: Protocol Framework | Write protocols | Guides | Core
```

File: doc/2/guides/write-protocols/entrypoint/index.md [hunk 17]
```diff
@@ -2,6 +2,12 @@
 code: false
 type: branch
 order: 300
-title: Protocol Entrypoint
-description: Protocol Entrypoint Reference
+title:  Protocol Entrypoint | Write protocols | Guides | Core
```
Comment (alexandrebouthinon) [inline doc/2/guides/write-protocols/entrypoint/index.md:5]:
```suggestion
title: Protocol Entrypoint | Write protocols | Guides | Core
```

File: doc/2/guides/write-protocols/methods/broadcast/index.md [hunk 18]
```diff
@@ -1,9 +1,13 @@
 ---
 code: true
 type: page
-title: broadcast
+title:  broadcast | Write protocols | Guide
```
Comment (alexandrebouthinon) [inline doc/2/guides/write-protocols/methods/broadcast/index.md:4]:
```suggestion
title: broadcast | Write protocols | Guide
```

File: doc/2/guides/write-protocols/methods/disconnect/index.md [hunk 19]
```diff
@@ -1,9 +1,13 @@
 ---
 code: true
 type: page
-title: disconnect
+title:  disconnect | Write protocols | Guide
```
Comment (alexandrebouthinon) [inline doc/2/guides/write-protocols/methods/disconnect/index.md:4]:
```suggestion
title: disconnect | Write protocols | Guide
```

File: doc/2/guides/write-protocols/methods/index.md [hunk 20]
```diff
@@ -2,6 +2,10 @@
 code: false
 type: branch
 order: 200
-title: Protocol Methods
-description: Protocol Methods Reference
----
+title:  Write protocols | Guide
```
Comment (alexandrebouthinon) [inline doc/2/guides/write-protocols/methods/index.md:5]:
```suggestion
title: Write protocols | Guide
```

File: doc/2/guides/write-protocols/methods/init/index.md [hunk 21]
```diff
@@ -1,7 +1,12 @@
 ---
 code: true
 type: page
-title: init
+title:  init | Write protocols | Guide
```
Comment (alexandrebouthinon) [inline doc/2/guides/write-protocols/methods/init/index.md:4]:
```suggestion
title: init | Write protocols | Guide 
```

File: doc/2/guides/write-protocols/methods/leavechannel/index.md [hunk 22]
```diff
@@ -1,7 +1,12 @@
 ---
 code: true
 type: page
-title: leaveChannel
+title:  leaveChannel | Write protocols | Guide
```
Comment (alexandrebouthinon) [inline doc/2/guides/write-protocols/methods/leavechannel/index.md:4]:
```suggestion
title: leaveChannel | Write protocols | Guide
```

File: doc/2/guides/write-protocols/methods/notify/index.md [hunk 23]
```diff
@@ -1,7 +1,12 @@
 ---
 code: true
 type: page
-title: notify
+title:  Notify | Protocol Methods | Write protocols | Guide
```
Comment (alexandrebouthinon) [inline doc/2/guides/write-protocols/methods/notify/index.md:4]:
```suggestion
title: Notify | Protocol Methods | Write protocols | Guide
```

**Review by alexandrebouthinon — APPROVED** [review]
I don't know if these whitespaces were intended (if so ignore my suggestions). I think it should be better to test these changes on the preprod environment (https://next-docs.kuzzle.io). To do so, you'll need to update the targeted branch

**Pull request merged**